design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,144 @@
+
+## Alternatives Considered
+An other option is making each BackupRepository having its own repository maintenance job configuration.
+Each BackupRepository has its own storage capacity, so they may require different resource to maintain. 
+This can be done by adding a field to the BackupRepository CRD to specify the referenced ConfigMap.
+After discussing, we think there is no need to make the new configuration for each BackupRepository.
+The configuration should be specific for the repository maintenance job.
+
+## Compatibility
+v1.14 uses the `velero server` CLI's parameter to pass the repository maintenance job configuration.
+If the ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` doesn't exist, those parameters are still respected.
+If the ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` exists, the ConfigMap provided parameters override the CLI parameters.
+
+## Design
+This design introduces a new ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` as the source of the repository maintenance job configuration. 
+If the `repo-maintenance-job-config` doesn't exist, the `velero server` parameters related to the repository maintenance job are used.
+
+**Notice**
+* Velero doesn't own this ConfigMap. If the user wants to customize the repository maintenance job, the user needs to create this ConfigMap.
+* Velero reads this ConfigMap content at start, so the ConfigMap change after the Velero server start cannot be reflected. To make the change take affect, please restart the Velero Deployment pod.
+
+### Structure
+The data structure for ```repo-maintenance-job-config``` is as below:
+```go
+type Configs struct {
+	// LoadAffinity is the config for data path load affinity.
+	LoadAffinity []*LoadAffinity `json:"loadAffinity,omitempty"`    
+
+    // How many completed maintenance jobs should be kept.
+    KeepLatestJobsCount int    `json:"keepLatestJobsCount,omitempty"`
+
+    // The repository maintenance job CPU request setting
+	CPURequest          string `json:"cpuRequest,omitempty"`
+
+    // The repository maintenance job memory request setting
+	MemRequest          string `json:"memRequest,omitempty"`
+
+    // The repository maintenance job CPU limit setting
+	CPULimit            string `json:"cpuLimit,omitempty"`
+
+    // The repository maintenance job memory limit setting
+	MemLimit            string `json:"memLimit,omitempty"`
+}
+
+type LoadAffinity struct {
+    // NodeSelector specifies the label selector to match nodes
+    NodeSelector metav1.LabelSelector `json:"nodeSelector"`
+}
+```
+
+The `LoadAffinity` structure is reused from design [node-agent affinity configuration](Implemented/node-agent-affinity.md).
+Users may want to set different LoadAffinity configurations according to different conditions (i.e., for different storages represented by StorageClass, CSI driver, etc.), so we define ```loadAffinity``` as an array.
```

Comment:
This configuration is only related to the repository, so why users want to set different configurations according to storageclass and CSI driver? And what storageclass does it mean? (the storageclass for the volume to be backed up?)